pkg/catalog/endpoint.go

```diff
@@ -36,3 +36,49 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	}
 	return endpoints, nil
 }
+
+// ListAllowedEndpointsForService returns only those endpoints for a service that belong to the allowed outbound service accounts
+// for the given downstream identity
+func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity service.K8sServiceAccount, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+	var outboundEndpoints []endpoint.Endpoint
+	outboundEndpoints, err := mc.ListEndpointsForService(upstreamSvc)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking  up endpoints for service %s with downstream identity %s", upstreamSvc, downstreamIdentity)
+		return nil, err
+	}
+	log.Trace().Msgf("Outbound endpoints for service %s with downstream identity %s: %v", upstreamSvc, downstreamIdentity, outboundEndpoints)
+
+	destSvcAccounts, err := mc.ListAllowedOutboundServiceAccounts(downstreamIdentity)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking up outbound service accounts for downstream identity %s", downstreamIdentity)
+		return nil, err
+	}
+
+	var allowedEndpoints []endpoint.Endpoint
+	for _, destSvcAccount := range destSvcAccounts {
+		podEndpoints := mc.listEndpointsforIdentity(destSvcAccount)
+		for _, ep := range outboundEndpoints {
+			for _, podIP := range podEndpoints {
+				if ep.IP.Equal(podIP.IP) {
+					allowedEndpoints = append(allowedEndpoints, ep)
```

Comment:
Can we just use `set.NewSetFromSlice()` and do `set1.Intersect(set2)`?

Reply:
We cannot use a set here cause `endpoint.Endpoint` has an unhashable type (`net.IP`)